artichoke-backend/src/extn/core/env/env.rb

```diff
@@ -52,63 +52,94 @@ def include?(name)
     to_h.key?(name)
   end
 
+  def invert
+    to_h.invert
+  end
+
   def key(value)
     to_h.key(value)
   end
 
+  def keys
+    to_h.keys
+  end
+
   def length
     to_h.length
   end
 
-  def size
-    to_h.size
+  def rehash
+    nil
   end
 
-  def keys
-    to_h.keys
-  end
+  def shift
+    envs = to_h
 
-  def rehash
-    nil
+    return nil if envs.nil? || envs.empty?
+
+    name, value = envs.shift
+
+    self[name] = nil
+    [name, value]
   end
 
-  def to_a
-    to_h.to_a
+  def size
+    to_h.size
   end
 
-  def to_s
-    'ENV'
+  def replace(hash)
```

Comment:
looks like this method ended up unalphabetized

Reply:
I reordered ENV's methods